### PR TITLE
[@types/chartjs-plugin-crosshair] Create new type definitions.

### PR DESCRIPTION
**Pull request by danmana**
[@types/chart.js] Allow extension of IteractionMode (backwards compatible changes)

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
[@types/chart.js] Allow extension of IteractionMode (backwards compatible changes)

### CODE REVIEW

**Comment by sandersn** [comment]
This change looks fine to me, but I don't like merging changes to such a widely used package like chart.js without some authors signing off.

@anuti @FabienLavocat @KentarouTakeda @larrybahr @mernen @josefpaij @guillaume-ro-fr @archy-bold @braincore @frabnt @alexdor @mahnuh @Conrad777 @adripanico @wertzui @lekoaf @ElianCordoba @indigolain @ricardo-mello @rnicholus @mrjack88 @canoceto @nobu222 @Marcoru97 @tonybadguy - 
Can you take a look?

If you haven't seen the interface-registry trick, it allows type aliases to be expanded by plugins, yet type checking will still be able to treat the type alias as complete -- with the original members for people without the plugin, but with additional members for people with the plugin.

If nobody from chart.js shows up in a few days to object, I think this change is fine to merge regardless.

File: types/chart.js/index.d.ts [hunk 1]
```diff
@@ -157,7 +157,20 @@ declare namespace Chart {
...
-    type InteractionMode = 'point' | 'nearest' | 'single' | 'label' | 'index' | 'x-axis' | 'dataset' | 'x' | 'y';
+    // Allow extending the IteractionMode type alias
+    // see https://github.com/microsoft/TypeScript/issues/28078#issuecomment-432339564
+    interface InteractionModeRegistry {
+        'point': 'point';
+        'nearest': 'nearest';
+        'single': 'single';
+        'label': 'label';
+        'index': 'index';
+        'xaxis': 'xaxis';
```
Comment (lekoaf) [inline types/chart.js/index.d.ts:168]: Why did you change this from `x-axis` to `xaxis`?

File: types/chart.js/index.d.ts [hunk 2]
```diff
@@ -157,7 +157,20 @@ declare namespace Chart {
...
+        'dataset': 'dataset';
+        'x': 'x';
+        'y': 'y';
+    }
+    type InteractionMode = InteractionModeRegistry[keyof InteractionModeRegistry];
```
Comment (lekoaf) [inline types/chart.js/index.d.ts:173]: Will this produce the exact same result as the row you removed?

**Review by lekoaf — CHANGES_REQUESTED** [review]

File: types/chart.js/index.d.ts (hunk 1 above)
Comment (danmana) [reply types/chart.js/index.d.ts:168]: You are right, this was a mistake. Will fix it soon

File: types/chart.js/index.d.ts (hunk 2 above)
Comment (danmana) [reply types/chart.js/index.d.ts:173]: Yes, this will produce the exact result/typing as the row that was removed, just now it's also extensible.

**Review by lekoaf — APPROVED** [review]
